azuread/data_client_config.go

```diff
@@ -26,11 +26,6 @@ func dataClientConfig() *schema.Resource {
 				Computed: true,
 			},
 
-			"subscription_id": {
```

Comment:
I think it would be best to deprecate this and remove it from all documentation & examples so we don't break people's config - we can remove it in 1.0

Reply:
Thanks, that's a good shout. I've reworked it so that for existing configurations, where a subscription ID is configured for the provider, it's returned by `data.azuread_client_config`. For newer configurations where it's omitted, the data source attribute is a blank string.